pkg/rbac/manifests.go

```diff
@@ -30,33 +30,51 @@ import (
 
 // ManifestOptions represent options for generating the RBAC manifests.
 type ManifestOptions struct {
-	InputDir  string
-	OutputDir string
-	Name      string
-	Labels    map[string]string
+	InputDir       string
+	OutputDir      string
+	RoleFile       string
+	BindingFile    string
+	Name           string
+	ServiceAccount string
+	Namespace      string
+	Labels         map[string]string
 }
 
 // SetDefaults sets up the default options for RBAC Manifest generator.
 func (o *ManifestOptions) SetDefaults() {
 	o.Name = "manager"
 	o.InputDir = filepath.Join(".", "pkg")
 	o.OutputDir = filepath.Join(".", "config", "rbac")
+	o.ServiceAccount = "default"
```

Comment:
This may be problematic, since it will override it even I have specified something.
We should default only when it's empty.
I think we should check it for all these 5 fields.

Reply:
As of now, it is being used directly on an empty object so there is no issue: [link](https://github.com/kubernetes-sigs/controller-tools/blob/master/cmd/controller-gen/main.go#L60-L61).

What you suggest is more like a `FillDefaults` than a `SetDefaults`. I agree with you that it would be a better approach as in a future it may be used in a non-empty object. However I think that changing that behaviour is not responsability of this PR. I will open a new Issue to track that change.

Reply:
Fair enough.
Thanks for opening the issue.

Reply:
Hmm... In our current code, `SetDefaults` happens before the fields are set by the flags.
IIRC in k/k we usually set fields using flags first then do the defaulting.